### ISSUE TITLE
how to use yolov4-tiny_contrastive.cfg?

### ISSUE THREAD

**Issue by stephanecharette** [AlexeyAB/darknet#6892]
- What is the new config file `yolov4-tiny_contrastive.cfg`?
- When do we use it?
- How do we use it?

I see the `filters=...` on line 413 is:  `(number_of_classes + 5) * 9`.  (The mask on line 419 contains 9 mask indexes, 0-8.)

So with the network I'm training with 1 class, I'm using `filters=54`.

But when I try to train with it, I get a segfault after 1017 iterations:

```
v3 (iou loss, Normalizer: (iou: 0.07, cls: 1.00) Region 52 Avg (IOU: 0.810298, GIOU: 0.805515), Class: 0.990291, Obj: 0.720371, No Obj: 0.000382, .5R: 0.987805, .75R: 0.804878, count: 82, class_loss = 2.362441, iou_loss = 170.592484, total_loss = 172.954926 
Command terminated by signal 11

> dmesg | grep darknet
[2482552.554782] traps: darknet[9947] general protection ip:5624934fa568 sp:7f35b68fa278 error:0 in darknet[562493468000+3aa000]
```

**Comment by stephanecharette** [comment]
@AlexeyAB can you comment on this one?  Thanks.

**Comment by AlexeyAB** [comment]
@stephanecharette 

In general Contrastive learning (Contrastive-loss is an extended Triple-loss) is suitable for un-supervised learning and for multi-camera object tracking.
I just currently don't have enough time to implement it well to get top results.

This post is about this: https://github.com/AlexeyAB/darknet/issues/6004



There is experimental YOLOv4-tiny-contrastive model that is trained on MSCOCO:

* cfg: https://raw.githubusercontent.com/AlexeyAB/darknet/master/cfg/yolov4-tiny_contrastive.cfg

* weights: https://github.com/AlexeyAB/darknet/releases/download/darknet_yolo_v4_pre/yolov4-tiny_contrastive_last.weights


`./darknet detector demo cfg/coco.data cfg/yolov4-tiny_contrastive.cfg yolov4-tiny_contrastive_last.weights test.avi -out_filename out_test.avi -ext_output`
